From 58383b3f7b8467f6c6c054c43d82517fce0436cb Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Pablo=20Correa=20G=C3=B3mez?= <ablocorrea@hotmail.com>
Date: Thu, 4 May 2023 00:03:42 +0200
Subject: [PATCH] gnome-languages: set UTF-8 prefix, if the locale is UTF-8 but
 not explicit

This fixes an incompatibility between muslc and glibc, where muslc
considers locales by default UTF-8, while glibc considers them ISO.
This is a real problem when for example, running flatpaks under
alpinelinux and postmarketOS.

Ref https://gitlab.alpinelinux.org/alpine/aports/-/issues/14851
---
 libgnome-desktop/gnome-languages.c | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/libgnome-desktop/gnome-languages.c b/libgnome-desktop/gnome-languages.c
index adfc4eee..8a0485c8 100644
--- a/libgnome-desktop/gnome-languages.c
+++ b/libgnome-desktop/gnome-languages.c
@@ -364,7 +364,14 @@ add_locale (const char *language_name,
         language_name_get_codeset_details (language_name, NULL, &is_utf8);
 
         if (is_utf8) {
-                name = g_strdup (language_name);
+                /* If the locale is UTF-8, but it's not explicit in the name,
+                 * append it to avoid compatibility issues, e.g: muslc
+                 * defaults to UTF-8 when missing, while glibc defaults to ISO
+                 */
+                if (strchr (language_name, '.') == NULL)
+                        name = g_strdup_printf ("%s.UTF-8", language_name);
+                else
+                        name = g_strdup (language_name);
         } else if (utf8_only) {
 
                 if (strchr (language_name, '.'))
-- 
GitLab

From 7dbe267c95fea0ea7f720a0b67496eb493e282e5 Mon Sep 17 00:00:00 2001
From: velsinki <112010-velsinki@users.noreply.gitlab.gnome.org>
Date: Tue, 28 Nov 2023 02:03:54 +0000
Subject: [PATCH] thumbnail: Fix typo in G_TYPE_CHECK_INSTANCE_TYPE

---
 libgnome-desktop/gnome-desktop-thumbnail.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgnome-desktop/gnome-desktop-thumbnail.h b/libgnome-desktop/gnome-desktop-thumbnail.h
index a4bb63dd..33ac083f 100644
--- a/libgnome-desktop/gnome-desktop-thumbnail.h
+++ b/libgnome-desktop/gnome-desktop-thumbnail.h
@@ -48,7 +48,7 @@ typedef enum {
 #define GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY		(gnome_desktop_thumbnail_factory_get_type ())
 #define GNOME_DESKTOP_THUMBNAIL_FACTORY(obj)	(G_TYPE_CHECK_INSTANCE_CAST ((obj), GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY, GnomeDesktopThumbnailFactory))
 #define GNOME_DESKTOP_THUMBNAIL_FACTORY_CLASS(klass)	(G_TYPE_CHECK_CLASS_CAST ((klass), GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY, GnomeDesktopThumbnailFactoryClass))
-#define GNOME_DESKTOP_IS_THUMBNAIL_FACTORY(obj)		(G_TYPE_INSTANCE_CHECK_TYPE ((obj), GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY))
+#define GNOME_DESKTOP_IS_THUMBNAIL_FACTORY(obj)		(G_TYPE_CHECK_INSTANCE_TYPE ((obj), GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY))
 #define GNOME_DESKTOP_IS_THUMBNAIL_FACTORY_CLASS(klass)	(G_TYPE_CLASS_CHECK_CLASS_TYPE ((klass), GNOME_DESKTOP_TYPE_THUMBNAIL_FACTORY))
 
 typedef struct _GnomeDesktopThumbnailFactory        GnomeDesktopThumbnailFactory;
-- 
GitLab

